pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCudaBase.hpp

```diff
@@ -259,17 +270,18 @@ class TNCudaBase : public TensornetBase<PrecisionT, Derived> {
             /* void * */ static_cast<void *>(dummy_device_data.getData()),
             /* const int64_t *tensorModeStrides */ nullptr,
             /* const int32_t immutable */ 1,
-            /* const int32_t adjoint */ adjoint,
+            /* const int32_t adjoint */ 0,
             /* const int32_t unitary */ 1,
             /* int64_t * */ &id));
         if (!gate_matrix.empty()) {
             auto gate_key = std::make_pair(opName, par);
             std::vector<CFP_t> matrix_cu =
                 cuUtil::complexToCu<ComplexT>(gate_matrix);
             gate_cache_->add_gate(static_cast<std::size_t>(id), gate_key,
-                                  matrix_cu);
+                                  matrix_cu, adjoint);
         } else {
-            gate_cache_->add_gate(static_cast<std::size_t>(id), opName, par);
+            gate_cache_->add_gate(static_cast<std::size_t>(id), opName, par,
```

Comment:
Could we get the matrix (or inverse) from `gates/Gates.hpp`, then pass it as a matrix (after casting if necessary)?

Reply:
Pass adjoint as an argument here is necessary but as you said we can store the inversed gate data in the gate/Gates.hpp and access it directly, which can avoid the transpose operations. Have added `TODO` in the `TNCudaGateCache.hpp`.